controllers/eventingauth_controller.go

```diff
@@ -102,53 +108,56 @@ func (r *eventingAuthReconciler) Reconcile(ctx context.Context, req kcontrollerr
-			if createAppErr != nil {
-				logger.Error(createAppErr, "Failed to create application in IAS")
-				if err := r.updateEventingAuthStatus(ctx, &cr, eamapiv1alpha1.ConditionApplicationReady, createAppErr); err != nil {
-					return kcontrollerruntime.Result{}, err
-				}
-				return kcontrollerruntime.Result{}, createAppErr
-			}
-			logger.Info("Successfully created application in IAS")
-			r.existingIasApplications[cr.Name] = iasApplication
-		}
-		cr.Status.Application = &eamapiv1alpha1.IASApplication{
-			Name: cr.Name,
-			UUID: iasApplication.GetId(),
-		}
-		if err := r.updateEventingAuthStatus(ctx, &cr, eamapiv1alpha1.ConditionApplicationReady, nil); err != nil {
-			return kcontrollerruntime.Result{}, err
-		}
 
-		logger.Info("Creating application secret on SKR")
-		appSecret, createSecretErr := skrClient.CreateSecret(ctx, iasApplication)
-		if createSecretErr != nil {
-			logger.Error(createSecretErr, "Failed to create application secret on SKR")
-			if err := r.updateEventingAuthStatus(ctx, &cr, eamapiv1alpha1.ConditionSecretReady, createSecretErr); err != nil {
+	iasApplication, appExists := r.existingIasApplications[cr.Name]
+	if !appExists {
+		var createAppErr error
+		logger.Info("Creating application in IAS")
+		iasApplication, createAppErr = r.iasClient.CreateApplication(ctx, cr.Name)
+		if createAppErr != nil {
+			logger.Error(createAppErr, "Failed to create application in IAS")
+			if err := r.updateEventingAuthStatus(ctx, &cr, eamapiv1alpha1.ConditionApplicationReady, createAppErr); err != nil {
 				return kcontrollerruntime.Result{}, err
 			}
 			return kcontrollerruntime.Result{}, createAppErr
 		}
-		logger.Info("Successfully created application secret on SKR")
-
-		// Because the application secret is created on the SKR, we can delete it from the cache.
-		delete(r.existingIasApplications, cr.Name)
+		logger.Info("Successfully created application in IAS")
+		r.existingIasApplications[cr.Name] = iasApplication
+	}
+	cr.Status.Application = &eamapiv1alpha1.IASApplication{
+		Name: cr.Name,
+		UUID: iasApplication.GetID(),
+	}
+	if err := r.updateEventingAuthStatus(ctx, &cr, eamapiv1alpha1.ConditionApplicationReady, nil); err != nil {
+		return kcontrollerruntime.Result{}, err
+	}
 
-		cr.Status.AuthSecret = &eamapiv1alpha1.AuthSecret{
-			ClusterId:      cr.Name,
-			NamespacedName: fmt.Sprintf("%s/%s", appSecret.Namespace, appSecret.Name),
-		}
-		if err := r.updateEventingAuthStatus(ctx, &cr, eamapiv1alpha1.ConditionSecretReady, nil); err != nil {
+	logger.Info("Creating application secret on SKR")
+	appSecret, createSecretErr := skrClient.CreateSecret(ctx, iasApplication)
+	if createSecretErr != nil {
+		logger.Error(createSecretErr, "Failed to create application secret on SKR")
+		if err := r.updateEventingAuthStatus(ctx, &cr, eamapiv1alpha1.ConditionSecretReady, createSecretErr); err != nil {
 			return kcontrollerruntime.Result{}, err
 		}
+		return kcontrollerruntime.Result{}, createSecretErr
```

Comment:
## Note for the reviewers

In the context of creating a secret, returning a `createAppErr` here is a bug. I changed it to return the proper error  which is `createSecretErr`.